[PW-4978] outsource reusable notification code (#1)
Adyen/php-webhook-module: README.md

# Webhook Module for PHP

Adyen library for handling notification webhooks.

## Installation

> You can use [Composer](https://getcomposer.org/). Follow the [installation instructions](https://getcomposer.org/doc/00-intro.md) if you do not already have composer installed.

Reviewer: ```suggestion
You can use [Composer](https://getcomposer.org/). Follow the [installation instructions](https://getcomposer.org/doc/00-intro.md) if you do not already have Composer installed.
```

~~~~ bash
composer require adyen/php-webhook-module
~~~~

> In your PHP script, make sure you include the autoloader:

Reviewer: ```suggestion
In your PHP script, make sure to include the autoloader:
```

$notificationReceiver = new \Adyen\Webhook\Receiver\NotificationReceiver(new \Adyen\Webhook\Receiver\HmacSignature);

// Authorize notification
if (!$notificationReceiver->isAuthenticated(
    $request['notificationItems'][0]['NotificationRequestItem'],
    YOUR_MERCHANT_ACCOUNT,
    YOUR_NOTIFICATION_USERNAME,
    YOUR_NOTIFICATION_PASSWORD
)) {
    throw new AuthenticationException();
}

// Process each notification item
foreach ($request['notificationItems'] as $notificationItem) {
    // validate the notification
    if ($notificationReceiver->validateHmac($notificationItem, YOUR_HMAC_KEY)) {
       // save notification to your database
       $this->databaseService->saveNotification($notificationItem); 
    }
}

return new JsonResponse('[accepted]');
~~~~

#### Process notification to get new payment state:
~~~~ php 
$notificationItem = \Adyen\Webhook\Notification::createItem([
    'eventCode' => $notification['eventCode'],
    'success' => $notification['success']
]);

$processor = \Adyen\Webhook\Processor\ProcessorFactory::create(
    $notificationItem,
    $currentPaymentState,
    $this->logger
);

$newState = $processor->process();
~~~~

> NB: set `$currentPaymentState` to one of the values in `\Adyen\Webhook\PaymentStates`

Reviewer: What is NB here?

Author: nota bene

Reviewer: ok bene

Reviewer 2: I also didn't recognise it at first, does it make sense to change it to this below:

```suggestion
Nota Bene: set `$currentPaymentState` to one of the values in `\Adyen\Webhook\PaymentStates`
```
